APM > Configuration > .NET > Adding IIS environment variables (#17811)
DataDog/documentation: content/en/tracing/trace_collection/dd_libraries/dotnet-core.md

{{< /tabs >}}

#### IIS

> After installing the MSI, no additional configuration is needed to automatically instrument your IIS sites. If you'd like to set additional environment variables that will be inherited by all IIS sites, perform the following steps:

Reviewer: ```suggestion
After installing the MSI, no additional configuration is needed to automatically instrument your IIS sites. To set additional environment variables that are inherited by all IIS sites, perform the following steps: 
```

> 1. Open the Registry Editor, find the multi-string value called `Environment` in the `HKLM\System\CurrentControlSet\Services\WAS` key, and add environment variables one-per-line.

Reviewer: ```suggestion
1. Open the Registry Editor, find the multi-string value called `Environment` in the `HKLM\System\CurrentControlSet\Services\WAS` key, and add the environment variables, one per line.
```

2. Run the following commands to restart IIS:
   ```cmd
   net stop /y was
   net start w3svc
   # Also, start any other services that were stopped when WAS was shut down.
   ```

> For example, to add logs injection and runtime metrics, add the following lines to the value data:

Author: Yes it is, and yes I can move it under step 1. I've updated it for your feedback in dcebfcc6cb6642b47124189a96b29820dbe4fed9
